tests/expressions/tests.py

```diff
@@ -2333,6 +2333,34 @@ def test_resolve_output_field(self):
                 expr = Value(value)
                 self.assertIsInstance(expr.output_field, output_field_type)
 
+    def test_mixed_output_field_with_none_returns_field(self):
+        test_cases = [
+            (Value(3.14159) + Value(None), FloatField),
+            (Value(42) + Value(None), IntegerField),
+            (Value(Decimal("3.14")) + Value(None), DecimalField),
```

Comment:
Please add tests for `None` on the left-hand side:
```suggestion
            (Value(Decimal("3.14")) + Value(None), DecimalField),
            (Value(None) + Value(3.14159), FloatField),
            (Value(None) + Value(42), IntegerField),
            (Value(None) + Value(Decimal("3.14")), DecimalField),
```

Reply:
Sorry I was unwell for a bit, coming back to this now, let me work on it.

Reply:
Added.

Reply:
Exhaustive version of this test for comparison: 

```
    def test_mixed_output_field_with_none_returns_field(self):
        def generate_test_cases():
            test_values_and_fields = [
                (3.14159, FloatField),
                (42, IntegerField),
                (Decimal("3.14"), DecimalField),
            ]
            connectors = [
                Combinable.ADD,
                Combinable.SUB,
                Combinable.MUL,
                Combinable.DIV,
                Combinable.MOD,
                Combinable.POW
            ]
            test_cases = []

            for value, result_field in test_values_and_fields:
                for connector in connectors:
                    test_cases.append((Value(value)._combine(Value(None),
                                                             connector,
                                                             reversed=False),
                                       result_field))
                    test_cases.append((Value(value)._combine(Value(None),
                                                             connector,
                                                             reversed=True),
                                       result_field))

            return test_cases

        test_cases = generate_test_cases()

        for expression, result in test_cases:
            with self.subTest(expression):
                self.assertIsInstance(expression.output_field, result)
```